### PR TITLE
Add Neon EVM to preloaded chains + remove Fantom

### PR DESCRIPTION
**Pull request by onyb**
This PR does three things:
1. Adds Neon EVM Mainnet to preloaded networks.
2. Removes Fantom Opera from preloaded networks (see internal [Slack discussion](https://bravesoftware.slack.com/archives/C023VS4HJ6Q/p1690554894315509)).
3. Changed the order of preloaded networks based on TVL from https://defillama.com/chains/EVM (except Aurora).

Resolves https://github.com/brave/brave-browser/issues/31760
Resolves https://github.com/brave/brave-browser/issues/32085

Token list update adding Neon logo: https://github.com/brave/token-lists/pull/75

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Testing addition of Neon EVM

1. Neon EVM shows up on Portfolio.
2. Neon logo is visible
3. The price of `NEON` is accurate

### Testing removal of Fantom Opera

(steps)

1. Open a Brave Browser build that does not contain changes in this PR.
2. Visit matcha.xyz and connect Brave Wallet.
3. Switch to Fantom
4. Open a Brave Browser build that contains this PR. Make sure the two browsers share the same user prefs.
5. Visit `brave://settings/wallet/networks`
6. Fantom Opera should be added as a custom chain
7. Visit matcha.xyz again and connect Brave Wallet.
8. The connection should default to Fantom Opera network.

### CODE REVIEW

**Review by StephenHeaps — APPROVED** [review]

**Review by muliswilliam — APPROVED** [review]
Frontend LGTM

File: browser/brave_wallet/brave_wallet_service_unittest.cc [hunk 1]
```diff
@@ -2191,6 +2191,194 @@ TEST_F(BraveWalletServiceUnitTest, MigrateUserAssetsAddIsERC1155) {
...
+      "0xa");
+
+  EXPECT_TRUE(
+      GetPrefs()->GetBoolean(kBraveWalletCustomNetworksCeloMainnetMigrated));
+}
+
+TEST_F(BraveWalletServiceUnitTest, MigrateFantomMainnetAsCustomNetwork) {
+  /////////////////////////////////////////////////////////////////////////////
+  // CASE 1: Fantom is the selected network of some origin                    /
+  /////////////////////////////////////////////////////////////////////////////
```
Comment (supermassive) [inline browser/brave_wallet/brave_wallet_service_unittest.cc:2290]: Oneline comment like 
`// CASE 1: Fantom...`
should be enough

File: components/brave_wallet_ui/options/asset-options.ts [hunk 2]
```diff
@@ -23,6 +23,8 @@ export const getNetworkLogo = (chainId: string, symbol: string): string => {
     return 'chrome://erc-token-images/celo.png'
   if (chainId === BraveWallet.ARBITRUM_MAINNET_CHAIN_ID)
     return 'chrome://erc-token-images/arb.png'
+  if (chainId === BraveWallet.NEON_EVM_MAINNET_CHAIN_ID)
+    return 'chrome://erc-token-images/neon.png'
   if (chainId === AllNetworksOption.chainId)
     return AllNetworksOption.iconUrls[0]
```
Comment (supermassive) [inline components/brave_wallet_ui/options/asset-options.ts:30]: I see some svgs are removed, do they relate somehow to these pngs?
Should we cleanup something there?

Comment (onyb) [reply components/brave_wallet_ui/options/asset-options.ts:30]: The svgs were not used anywhere. All token logos should be used via `chrome://erc-token-images/*.png`. Neon logo was added here: https://github.com/brave/token-lists/pull/75

File: components/brave_wallet/browser/brave_wallet_service.cc [hunk 3]
```diff
@@ -1249,6 +1250,74 @@ void BraveWalletService::MigrateUserAssetsAddIsSpam(PrefService* prefs) {
...
+      const auto* chain_id_each = network_value.GetDict().FindString("chainId");
+      if (!chain_id_each) {
+        continue;
+      }
+
+      if (base::ToLowerASCII(*chain_id_each) == chain_id) {
+        return false;
+      }
+    }
+  }
```
Comment (supermassive) [inline components/brave_wallet/browser/brave_wallet_service.cc:1269]: nit: there is `CustomChainExists`

**Review by supermassive — APPROVED** [review]
LGTM

File: components/brave_wallet/browser/brave_wallet_service.cc (hunk 3 above)
Comment (onyb) [reply components/brave_wallet/browser/brave_wallet_service.cc]: https://github.com/brave/brave-core/pull/19574/commits/74afc0c520a943998b558e937568cc63aa3ea14f

**Pull request merged**